### PR TITLE
Refine Java section

### PR DESCRIPTION
**Pull request by meistermeier**
With different libraries for each level of abstraction available it would make a lot of sense to handle the Java menu item as a parent category.
This adds our own technologies as sub-items to the list but for now keeps the Java driver together with procedures, JDBC, JCypher, Grails plugin, neotypes and Hibernate OGM on the main page.
I would like to discuss if it might be better to move the third party content to a _third party_ sub-item, move procedures out of the whole _Java_ section and just keep the Java driver on the root page.

This PR represents also WIP to give the Neo4j Java Driver Spring Boot Starter and SDN-RX introduction-like documentation a place to live.

- [x] Introduce sub-sections
- [x] Move third-party libraries to sub-section
- [x] Move procedures documentation to sub-section
- [x] Add Neo4j Java Driver Spring Boot starter manual
- [x] Add Spring Data Neo4j RX manual

### CODE REVIEW

**Comment by meistermeier** [comment]
I could not bring the right CSS to the party and just hope that the new sub-menu items are ok-ish.

**Comment by meistermeier** [comment]
Applied the changes in `java.adoc` to `java-third-party.adoc`.

**Review by JMHReif — CHANGES_REQUESTED** [review]
The elements need adjusted on the menu_partial.erb so that everything is under the Java item. I think it should be something like below:
```
<li><a href="/developer/language-guides/">Drivers Overview</a></li>
                <li><a href="/developer/java/">Java</a></li>
                <ul>
                    <li><a href="/developer/java-procedures/">Procedures and Functions</a></li>
                    <li><a href="/developer/java-third-party/">Third-party libraries</a></li>
                    <li><a href="/developer/spring-data-neo4j/">Spring Framework</a></li>
                    <li><a href="/developer/java-driver-spring-boot-starter/">Java Driver Spring Boot Starter</a></li>
                    <li><a href="/developer/neo4j-ogm/">Neo4j Object Graph Mapper</a></li>
                    <li><a href="/developer/spring-data-neo4j/">Spring Data Neo4j</a></li>
                    <li><a href="/developer/spring-data-neo4j-rx/">Spring Data Neo4j RX</a></li>
                <ul>
```

File: language-guides/java-third-party/java-third-party.adoc [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+include::../_includes/community-driver-notice.adoc[]
+
+[#neo4j-jdbc]
+=== Using Neo4j Server with JDBC
+
+Being a Java developer you're probably familiar with JDBC as a way to work with relational databases, either directly or via abstractions like Spring's `JDBCTemplate` or MyBatis.
+Many tools for ETL, Report Generators, Business Intelligence, and data management also use JDBC drivers to interact with relational databases.
+
+As Cypher, like SQL, is a textual parametrizable query language that can return tabular results, it was possible for us to support the JDBC APIs and make a http://github.com/neo4j-contrib/neo4j-jdbc[Neo4j-JDBC^] driver available.
```
Comment (JMHReif) [inline language-guides/java-third-party/java-third-party.adoc]: typo: parameterizable

File: language-guides/java-third-party/java-third-party.adoc [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+| icon:book[] Docs | http://jcypher.iot-solutions.net/
+| icon:book[] Article | https://neo4j.com/blog/jcypher-focus-on-your-domain-model-not-how-to-map-it-to-the-database/
+|===
+
+[#neo4j-grails]
+=== Groovy & Grails: Neo4j Grails Plugin
+
+image::{img}grails.png[width=200,float=right]
+
+The goal of GORM for Neo4j is to provide a 'as-complete-as-possible' GORM implementation that maps domain classes and instances to the Neo4j nodespace. The following features are supported:
```
Comment (JMHReif) [inline language-guides/java-third-party/java-third-party.adoc:96]: Isn't GORM a GoLang ORM? How does that tie in with Groovy and Grails?

File: language-guides/java-third-party/java-third-party.adoc [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+* Marshalling from Neo4j Nodes to Groovy types and back again
+* Support for GORM dynamic finders, criteria and named queries
+* Session-managed transactions
+* Access to Neo4j's traversal capabilities
+* Access the Neo4j graph database in all flavours (Embedded, REST and HA)
```
Comment (JMHReif) [inline language-guides/java-third-party/java-third-party.adoc]: We should avoid using the term HA, as that was our legacy clustering technology. If it works with clusters, then let's just go with that wording.

**Comment by meistermeier** [comment]
Isn't the alignment correct?
https://github.com/neo4j-contrib/developer-resources/pull/164/files#diff-a7cb11564cb2fa8ffdef9ada0c5654cfR92-R100

File: language-guides/java-third-party/java-third-party.adoc (hunk 2 above)
Comment (meistermeier) [reply language-guides/java-third-party/java-third-party.adoc:96]: Nope it's Grail's object mapper https://gorm.grails.org/

File: language-guides/java-third-party/java-third-party.adoc (hunk 3 above)
Comment (meistermeier) [reply language-guides/java-third-party/java-third-party.adoc]: Thanks for the ping: It is now Bolt only.

File: language-guides/java-third-party/java-third-party.adoc (hunk 2 above)
Comment (JMHReif) [reply language-guides/java-third-party/java-third-party.adoc:96]: Ah, ok. I wasn't familiar with that term, so I did a quick Google search. The GoLang tool came up, so I guess the name overlaps with other things. :P

**Review by JMHReif — APPROVED** [review]

**Pull request merged**